lib/json_commenter.rb

```diff
@@ -0,0 +1,26 @@
+require "json"
+
+class JsonCommenter
+  def comment(comment)
+    current_comments.merge!(comment) if comment.is_a?(Hash)
+    yield
+  ensure
+    comment.each { |k, _| current_comments.delete(k) } if comment.is_a?(Hash)
```

Comment:
This will not work. Consider the case where one of the keys in `comment` override an existing key in `current_comments`. After the method call, the original value (and key) would be deleted from `current_comments`.

I think we need something more like this:
```rb
def comment(comment, &block)
  return block.call if !comment.is_a?(Hash)

  orig = current_comments.dup
  current_comments.merge!(comment)
  block.call
ensure
  current_comments.replace(orig)
end
```

We probably also need a test case.